Add > SP Cloud > Integrations Doc > Jenkins (#94)
turbot/steampipe-docs: docs/integrations/jenkins.md

---
title: Using Steampipe in Jenkins
sidebar_label: Jenkins
---

# Using Steampipe in Jenkins

> Jenkins provides a [hosted environment](https://www.jenkins.io/) in which you can build, test, and deploy software. This happens in a [Jenkins Pipeline](https://www.jenkins.io/doc/book/pipeline/). Let's install Steampipe through using the pipeline, then install a plugin and run a query.

Reviewer: ```suggestion
Jenkins provides a [hosted environment](https://www.jenkins.io/) in which you can build, test, and deploy software. This happens in a [Jenkins Pipeline](https://www.jenkins.io/doc/book/pipeline/). Let's use a pipeline to install Steampipe,  then install a plugin and run a query.
```

## Installing Steampipe in a Jenkins pipeline

To run scripts, you first create a `Jenkinsfile` which is a text file that contains the definition of a Jenkins Pipeline. Here's an example that installs Steampipe.

```
pipeline {
    agent any

    stages {
        stage("Install") {
            steps {
                sh "curl -s -L https://github.com/turbot/steampipe/releases/latest/download/steampipe_linux_amd64.tar.gz | tar -xzf -"
                echo "installed steampipe"
            }
        }
    }
}
```

## Running Steampipe in a Jenkins pipeline

> In order to run Steampipe commands. We will first install the [Hacker News](https://hub.steampipe.io/plugins/turbot/hackernews) plugin.

Reviewer: ```suggestion
In order to run Steampipe commands, we will first install the [Hacker News](https://hub.steampipe.io/plugins/turbot/hackernews) plugin.
```
